### ISSUE TITLE
[NEWS] Sandbox - kupujący może zmienić adres dostawy w zamówieniu / Sandbox - buyer can change shipping data

### ISSUE THREAD

**Issue by PrzemyslawLukanowski** [allegro/allegro-api#3730]
W drugiej połowie października udostępnimy kupującym możliwość zmiany adresu dostawy w zamówieniu. Klient będzie mógł zmienić dane dla zamówienia w statusie READY FOR PROCESSING, jeśli np. wcześniej podał nieprawidłowe informacje. Podobnie jak w przypadku anulowania zamówienia, będą musiały zostać spełnione odpowiednie warunki:

- sprzedający musi posiadać konto Firma,
- status realizacji zamówienia (fulfillment.status) jest inny niż: “PROCESSING”, “READY_FOR_SHIPMENT”, “SENT”;
- sprzedawca nie dodał numeru przesyłki do zamówienia,
- zmiana adresu możliwa jest w ciągu 3 dni od daty zakupu.

**Funkcjonalność udostępniliśmy już dzisiaj w naszym środowisku testowym**. W związku z tym dla:

- [GET /order/events](https://developer.allegro.pl/documentation/#operation/getOrderEventsUsingGET) - dodaliśmy nowy typ zdarzenia **“BUYER_MODIFIED**”, które otrzymasz, gdy kupujący zmieni adres;
- [GET /order/checkout-forms](https://developer.allegro.pl/documentation/#operation/getListOfOrdersUsingGET) i [GET /order/checkout-forms/{id}](https://developer.allegro.pl/documentation/#operation/getOrdersDetailsUsingGET) - w sekcji **delivery.address** dodaliśmy nowe pole **modifiedAt**, w którym zwracamy:
  - datę, kiedy kupujący zmienił dane w zamówieniu;
  - wartość null dla zamówień bez modyfikacji przez kupującego. 
Pole dodaliśmy także w środowisku produkcyjnym, ale do momentu udostępnienia funkcjonalności kupującym będzie tam zawsze wartość null.

Aby zmienić adres dostawy jako kupujący, wejdź w szczegóły zakupu w zakładce [Kupione ](https://allegro.pl.allegrosandbox.pl/moje-allegro/zakupy/kupione)i wybierz “Zmień” pod danymi odbiorcy.

-----

In the second half of October, we will allow buyers to change the address in the shipping form data. The customer will be able to change the shipping data for the order in the READY FOR PROCESSING status, if, for example, previously provided incorrect information. As in the order cancellation, certain conditions will have to be met:

- seller has a company account;
- order fulfillment status (“fulfillment.status”) is different than: PROCESSING, READY_FOR_SHIPMENT, SENT;
- seller has not added tracking number to the order.
- change of address is available within 3 days from the date of purchase.

**The feature is now available in our test environment.** Therefore, for:

- G[ET /order/events](https://developer.allegro.pl/documentation/#operation/getOrderEventsUsingGET) - we added a new event type **"BUYER_MODIFIED"**, which you will receive when the buyer changes the address;
- [GET /order/checkout-forms](https://developer.allegro.pl/documentation/#operation/getListOfOrdersUsingGET) and [GET /order/checkout-forms/{id}](https://developer.allegro.pl/documentation/#operation/getOrdersDetailsUsingGET) - in **delivery.address** section we added a new field - **modifiedAt**, in which we return:
  - the date when the buyer changed the data in the order;
  - null for orders without buyer modification.
We also added this field to the production environment, but until the functionality is available to buyers, we will always return null.

To change the shipping data as a buyer, go to purchase details in the [Your Purchases](https://allegro.pl.allegrosandbox.pl/moje-allegro/zakupy/kupione) tab and choose “Change” below the customer data.

**Comment by Onixarts** [comment]
Naciskam przycisk ZMIEŃ pod adresem, ale nic się nie dzieje. Przeglądarka nowy Edge.

**Comment by AureliuszBrussy** [comment]
@Onixarts Rzeczywiście, problem istnieje - zgłosiliśmy to już do zespołu odpowiedzialnego za funkcjonalność. Wrócę z odpowiedzią.

**Comment by AureliuszBrussy** [comment]
@Onixarts Problem został rozwiązany - możliwość edycji adresu powinna być już dostępna.

**Comment by Onixarts** [comment]
Edycja działa, choć czasami przycisk ZMIEŃ potrafi zniknąć przy odświeżaniu strony i pojawić się po ponownym odświeżeniu.
Zauważyłem też, że nie można zmieniać adresu odbioru w punkcie - tak ma być? Np. przy wyborze paczkomatu możliwość zmiany adresu widnieje tylko przy numerze telefonu kupującego - ale tutaj jednak nadal gumowy przycisk, nic się nie dzieje po jego naciśnięciu.

**Comment by AureliuszBrussy** [comment]
Tak, nie jest możliwe edytowanie adresu punktu dostawy, w takim przypadku Kupujący może dokonać tylko zmiany numeru telefonu. Zmiana numeru telefonu powinna być już możliwa - modal powinien się już wyświetlać.

**Comment by czlowiekpolar** [comment]
Brakuje nowego typu eventu w dokumentacji: https://developer.allegro.pl/documentation/#operation/getOrderEventsUsingGET

**Comment by MaciejFrackowiak** [comment]
@czlowiekpolar Opublikowaliśmy aktualizację dot. wprowadzenia tej funkcjonalności na środowisku produkcyjnym - https://github.com/allegro/allegro-api/issues/3831 - dlatego event nie został jeszcze dodany do dokumentacji.

**Comment by dszewczyk1** [comment]
> Tak, nie jest możliwe edytowanie adresu punktu dostawy, w takim przypadku Kupujący może dokonać tylko zmiany numeru telefonu. Powinieneś już móc zmienić numer telefonu - modal powinien się już wyświetlać.

W wersji Sandbox da się teraz zmienić wybrany paczkomat (nawet jest odpowiedni komunikat, że powiadomiono o tym sprzedawcę). Ale nie ma po tym śladu w API. Ani w dzienniku zdarzeń, ani w szczegółach zamówienia. 
Na Moje Allegro kupującego pojawia się zmieniony paczkomat.

**Comment by MaciejFrackowiak** [comment]
@dszewczyk1 To efekt dalszych testów funkcjonalności - oczywiście, gdy ostateczna wersja pojawi się na produkcji/Sandbox będziemy zwracać wszystkie przynależne eventy, szczegóły, etc. Obecnie wyłącznie dla zmiany danych adresowych są one generowane.
